evap/evaluation/tools.py

```diff
@@ -276,7 +283,9 @@ def next_row(self) -> None:
         self.cur_col = 0
         self.cur_row += 1
 
-    def write_row(self, vals: Iterable[str], style: str = "default") -> None:
+    def write_row(
+        self, vals: Iterable[CellValue | None], style: str | typing.Callable[[CellValue | None], str] = "default"
```

Comment:
Shouldn't this be generic over the type of the iterator item? Because if I give in some numbers and a function that expects numbers, I don't want typing to fail because I didn't handle the case where `style` is called with a string. Isn't it something like `TypeVar(..., bound=CellValue)`? We can also postpone this until it actually occurs somewhere though.

Reply:
added a typevar for this